src/api_server.h

```diff
@@ -44,6 +44,8 @@ class MetadataApiServer {
   ~MetadataApiServer();
 
  private:
+  friend class ApiServerTest;
+
```

Comment:
There is actually a bug in line 43 of `api_server.cc` — reverse iterators should still be incremented with `++`. Can you please change `--it` to `++it`?